### DISCUSSION TITLE
How to keep my project updated?

### DISCUSSION THREAD

**Discussion by itaied246** [kriasoft/react-starter-kit#1971] [Q&A]
I am developing node and react web app based on react-starter-kit.
As any project do, I override unnecessary code and files, and modify others as well.

Trying to keep up with the almost daily updates, I can't find a way to keep my project updated unless I resolve loads of conflicts every time.

The repository suggests to just merge to master, but it produces a lot of conflicts. Is there a quick and more reasonable way to do it?

[stack overflow question](http://stackoverflow.com/questions/40758229/how-to-keep-rsk-project-updated)

**Comment by koistya** [comment]
### Option A

After [generating a new project](https://github.com/kriasoft/react-starter-kit/generate) from this template, subscribe to React Starter Kit updates by watching all activities in this repository. Whenever a new PR is created, you would review modified files and the PR description. Then pull the latest version of the template to your local machine (`git pull`) and do a folder comparison using a tool such as [Beyond Compare](https://www.scootersoftware.com/) or similar, merging updates from `react-starter-kit` folder into your project.

![image](https://user-images.githubusercontent.com/197134/172364150-55da89ea-36b5-42f0-b1db-2492ec4aa8dc.png)

### Option B

If you don't mind having full `react-starter-kit` Git history in your project, you can pull and merge updates using Git:

```bash
# Create a new project folder from the template
$ git clone -o seed -b main --single-branch https://github.com/kriasoft/react-starter-kit.git example

# Add your Git repository as the origin
$ cd ./example
$ git remote add origin https://github.com/user/example.git
$ git push origin main -u


# Pull and merge updates from the template repository down the road
$ git checkout main
$ git fetch seed
$ git merge seed/main
```

And resolve conflicts using VSCode, WebStorm, Beyond Compare, or some other Git merge tool.

**Pro Tip**: Update your global Git config with the preferred merge tool:

- [How to resolve merge conflicts in Git](https://www.youtube.com/watch?v=xNVM5UxlFSA)
- [How to configure Beyond Compare as a Git merge tool](https://www.scootersoftware.com/support.php?zz=kb_vcs_osx)
- [How to configure VSCode as the a Git merge tool](https://stackoverflow.com/questions/44549733/how-to-use-visual-studio-code-as-the-default-editor-for-git-mergetool)
- [Git's mergetool documentation](https://git-scm.com/docs/git-mergetool)
